### PR TITLE
Refactoring 2 [API-1649]

### PR DESCRIPTION
**Pull request by yuce**
This PR adds the initial CLC command and implements the `plug.CommandContext` and `plug.ExecContext` interfaces.

The plugins for actual commands will be sent in a separate PR.

### CODE REVIEW

File: clc/cmd/command_context.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	return cc.isInteractive
+}
+
+func (cc *CommandContext) SetCommandHelp(long, short string) {
+	if long != "" {
+		cc.Cmd.Long = long
+	}
+	if short != "" {
+		cc.Cmd.Short = short
+	}
```
Comment (emreyigit) [inline clc/cmd/command_context.go:87]: What if both arguments are empty?

File: clc/cmd/command_context.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/mk"
+)
+
+type CommandContext struct {
+	Cmd           *cobra.Command
+	Vpr           *viper.Viper
+	stringValues  map[string]*string
```
Comment (emreyigit) [inline clc/cmd/command_context.go:15]: Just wondering. What's the value of `*string` or same for other primitives in the map?

Comment (yuce) [reply clc/cmd/command_context.go:15]: Using primitives would copy their values, so Cobra wouldn't be able to change them.

File: clc/cmd/command_context.go (hunk 1 above)
Comment (yuce) [reply clc/cmd/command_context.go:87]: It wouldn't display the help string for that command.

File: clc/cmd/exec_context.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	if ec.ci != nil {
+		return ec.ci, nil
+	}
+	client, err := ec.ExecuteBlocking(ctx, "Connecting to the cluster", func(ctx context.Context) (any, error) {
+		return ec.clientFn(ctx)
+	})
+	if err != nil {
+		return nil, err
+	}
+	ec.ci = hazelcast.NewClientInternal(client.(*hazelcast.Client))
```
Comment (emreyigit) [inline clc/cmd/exec_context.go:101]: I couldn't understand here. What's happening?

**Review by emreyigit — APPROVED** [review]

**Pull request merged**